java/patterns/README.md

```diff
@@ -0,0 +1,12 @@
+# Restate Sample Patterns
+
+A collections of useful patterns in distributed applications, and how to
+implement them with [Restate](https://github.com/restatedev/restate).
+This is a continuously evolving list.
+
+All patterns are described in one file and have a comment block at the top
+that explains them.
+
+### List of Patterns
+
+- [**Compensations**](src/main/java/dev/restate/patterns/Compensations.java)
```

Comment:
Are the patterns meant to be run locally? Or is it more like a reference? Otherwise, it would be nice to have an example invoke cmd to copy over. And a little explanation on how to try out the compensation.

Reply:
The patterns are not meant to be run locally atm. They should compile though. If you think being able to run them adds value, then we can add it as a follow-up.

Reply:
I'm fine with not running them, but maybe we should mention that?